File: lib/models/log_entry.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'log_entry.freezed.dart';
part 'log_entry.g.dart';

enum AppLogLevel { debug, info, success, warning, error }

@freezed
abstract class LogEntry with _$LogEntry {
  const LogEntry._();

  const factory LogEntry({
    required DateTime timestamp,
    required AppLogLevel level,
    required String message,
    String? source,
  }) = _LogEntry;

  factory LogEntry.fromJson(Map<String, dynamic> json) =>
      _$LogEntryFromJson(json);

  String get displayLevel => level.name.toUpperCase();
}

File: app/src/main/java/io/github/rustnomicon/sourceinstaller/data/model/LogEntry.kt
package io.github.rustnomicon.sourceinstaller.data.model

enum class AppLogLevel { Debug, Info, Success, Warning, Error }

/**
 * Port of the Flutter `LogEntry` model. Timestamp is epoch milliseconds.
 */
data class LogEntry(
    val timestampMillis: Long,
    val level: AppLogLevel,
    val message: String,
    val source: String? = null,
) {
    val displayLevel: String get() = level.name.uppercase()
}
